### PR TITLE
customcluster: support more ha config

### PR DESCRIPTION
**Pull request by Xieql**
**What type of PR is this?**

/kind feature

**What this PR does / why we need it**:
part of https://github.com/kurator-dev/kurator/issues/128

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: pkg/apis/infra/v1alpha1/cluster.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ type ControlPlaneConfig struct {
 	// CertSANs sets extra Subject Alternative Names for the API Server signing cert.
 	// +optional
 	CertSANs []string `json:"certSANs,omitempty"`
+	// LBDomainName is a variable used to set the endpoint for a Kubernetes cluster
+	// when a load balancer (LB) is enabled. The default value is "lb-apiserver.kubernetes.local".
+	LBDomainName string `json:"lbDomainName,omitempty"`
```
Comment (zirain) [inline pkg/apis/infra/v1alpha1/cluster.go:68]: prefer `LoadBalancerDomainName` or `LoadBalancer.DomainName`

Comment (Xieql) [reply pkg/apis/infra/v1alpha1/cluster.go]: rename to LoadBalancerDomainName

File: pkg/apis/infra/v1alpha1/cluster.go [hunk 2]
```diff
@@ -63,6 +63,9 @@ type ControlPlaneConfig struct {
 	// CertSANs sets extra Subject Alternative Names for the API Server signing cert.
 	// +optional
 	CertSANs []string `json:"certSANs,omitempty"`
+	// LoadBalancerDomainName is a variable used to set the endpoint for a Kubernetes cluster
+	// when a load balancer (LB) is enabled. The default value is "lb-apiserver.kubernetes.local".
+	LoadBalancerDomainName string `json:"loadBalancerDomainName,omitempty"`
```
Comment (hzxuzhonghu) [inline pkg/apis/infra/v1alpha1/cluster.go:68]: I think we should not dedault it to "lb-apiserver.kubernetes.local". 

It is common for user to have no domain, can we set it to a public address

Comment (Xieql) [reply pkg/apis/infra/v1alpha1/cluster.go:68]: yes，we can do it

**Review by hzxuzhonghu — APPROVED** [review]
/lgtm

**Pull request merged**